packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -1029,8 +1029,13 @@ export class Pipeline extends PipelineBase {
           ? action.actionProperties.resource.env.region
           : action.actionProperties.region;
         const pipelineStack = Stack.of(this);
+
+        // If the token is unresolved, we let Stack construct to generate the stack name for us.
+        const stackName = Token.isUnresolved(pipelineStack.stackName)
+          ? undefined
+          : `${pipelineStack.stackName}-support-${targetAccount}`;
```

Comment:
From my understanding, it seems like if the stack name is a token, we're deferring the generation of the stack name to the Stack construct's creation. The only difference I can see between here and that new `generateStackName()` is that this one uses suffix `targetAccount` and the other uses `props.region`. It also seems like it's making the token check now as well. Is backwards compatibility the only thing stopping us from moving this logic to that one central method?

Reply:
`pipelineStack.stackName` is a token, so if you use it as part of the stack name generation, the generated stack name would fail a subsequent check here https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/core/lib/stack.ts#L477

The reason we don't use the same function between here and the `generateStackName` is partly due to backward compatibility issue.